py/mkrules.cmake

```diff
@@ -0,0 +1,108 @@
+# CMake fragment for MicroPython rules
+
+set(MPY_PY_QSTRDEFS "${MPY_PY_DIR}/qstrdefs.h")
+set(MPY_GENHDR_DIR "${CMAKE_BINARY_DIR}/genhdr")
+set(MPY_MPVERSION "${MPY_GENHDR_DIR}/mpversion.h")
+set(MPY_MODULEDEFS "${MPY_GENHDR_DIR}/moduledefs.h")
+set(MPY_QSTR_DEFS_LAST "${MPY_GENHDR_DIR}/qstr.i.last")
+set(MPY_QSTR_DEFS_SPLIT "${MPY_GENHDR_DIR}/qstr.split")
+set(MPY_QSTR_DEFS_COLLECTED "${MPY_GENHDR_DIR}/qstrdefs.collected.h")
+set(MPY_QSTR_DEFS_PREPROCESSED "${MPY_GENHDR_DIR}/qstrdefs.preprocessed.h")
+set(MPY_QSTR_DEFS_GENERATED "${MPY_GENHDR_DIR}/qstrdefs.generated.h")
+
+target_sources(${MICROPYTHON_TARGET} PRIVATE
+    ${MPY_MPVERSION}
+    ${MPY_QSTR_DEFS_GENERATED}
+)
+
+# Command to force the build of another command
+
+add_custom_command(
+    OUTPUT FORCE_BUILD
+    COMMENT ""
+    COMMAND echo -n
+)
+
+# Generate mpversion.h
+
+add_custom_command(
+    OUTPUT ${MPY_MPVERSION}
+    COMMAND ${CMAKE_COMMAND} -E make_directory ${MPY_GENHDR_DIR}
+    COMMAND python3 ${MPY_DIR}/py/makeversionhdr.py ${MPY_MPVERSION}
```

Comment:
Please use [FindPython3](https://cmake.org/cmake/help/latest/module/FindPython3.html), i.e. `Python3 REQUIRED COMPONENTS Interpreter`.

In a virtualenv, the Python 3 interpreter might not have a "3" suffix, and this leads to build failure.  Please use `${Python3_EXECUTABLE}` instead of hardcoding the name, and this should work in all situations.